core/src/test/java/io/kestra/core/runners/RunContextTest.java

```diff
@@ -112,14 +113,15 @@ void inputsLarge() throws TimeoutException {
     void variables() throws TimeoutException {
         Execution execution = runnerUtils.runOne(null, "io.kestra.tests", "return");
 
-        assertThat(execution.getTaskRunList(), hasSize(3));
+        assertThat(execution.getTaskRunList(), hasSize(4));
 
         assertThat(
             ZonedDateTime.from(ZonedDateTime.parse((String) execution.getTaskRunList().get(0).getOutputs().get("value"))),
             ZonedDateTimeMatchers.within(10, ChronoUnit.SECONDS, ZonedDateTime.now())
         );
         assertThat(execution.getTaskRunList().get(1).getOutputs().get("value"), is("task-id"));
         assertThat(execution.getTaskRunList().get(2).getOutputs().get("value"), is("return"));
+        assertThat((String) execution.getTaskRunList().get(3).getOutputs().get("value"), containsString("toto"));
```

Comment:
Please add an infinite loop test and assert that you get the result that preceeds the infinite loop turn
(var1: {{var2}}, var2: {{var3}}, var3: {{var1}})
=> render({{var1 | replace({'3': '4'})}}) => {{var4}}

Reply:
you can change the test as it's purely logic and not easily understandable but you get it